### PR TITLE
ISPN-3550 Migrate Cassandra cache store to the 6.0 cachestore API

### PR DESCRIPTION
**Pull request by jmarkos**
Some open problems:
1. should we use a dedicated Log class for logging?
2. process() is not parallel
3. the old version of the store allowed to configure the readConsistencyLevel and writeConsistencyLevel - but there are a lot more configuration options for the driver, how do we decide which ones to expose in the store configuration?
4. there are some errors from the cassandra daemon after the tests finish
5. Cassandra can store byte arrays directly, so I didn't use the Key2stringmapper

The tests require a small change in the infinispan repository: https://github.com/infinispan/infinispan/pull/3941

### CODE REVIEW

**Comment by tristantarrant** [comment]
Yes it needs its own logger with its own messages for internationalization.
And we want to at least let the users select their desired consistency level

**Comment by jmarkos** [comment]
Updated.

**Comment by jmarkos** [comment]
Updated - trace/debug logs are inline, error ones are in a custom log.

File: src/main/java/org/infinispan/persistence/cassandra/CassandraStore.java [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+         entryTable = configuration.entryTable();
+         writeStatement = session.prepare("INSERT INTO " + entryTable + " (key, value, metadata) VALUES (?, ?, ?) USING TTL ?");
+         deleteStatement = session.prepare("DELETE FROM " + entryTable + " WHERE key=?");
+         selectStatement = session.prepare("SELECT value, metadata FROM " + entryTable + " WHERE key=?");
+         containsStatement = session.prepare("SELECT key FROM " + entryTable + " WHERE key=?");
+         selectAllStatement = session.prepare("SELECT key, value, metadata FROM " + entryTable);
+         sizeStatement = session.prepare("SELECT count(*) FROM " + entryTable);
+         clearStatement = session.prepare("TRUNCATE " + entryTable);
+      } catch (Exception e) {
+         log.errorCommunicating(e);
```
Comment (tristantarrant) [inline src/main/java/org/infinispan/persistence/cassandra/CassandraStore.java]: Best way to do this is to change the "void errorCommunicating(e)" method signateu to return a CacheConfigurationException and then you can "throw log.errorCommunicating(e)". Same for all others

**Comment by tristantarrant** [comment]
And.... it's in ! Thanks a lot @jmarkos

**Pull request closed without merging**